### PR TITLE
Deprecate OIDC

### LINKED ISSUE #5061: Deprecate embedded OIDC support

### PR DESCRIPTION
**Pull request by arielshaqed**
Closes #5061.

### CODE REVIEW

**Comment by arielshaqed** [comment]
@Isan-Rivkin PTAL!

File: api/swagger.yml [hunk 1]
```diff
@@ -610,9 +610,13 @@ components:
           type: string
           enum: [initialized, not_initialized, comm_prefs_done]
         oidc_enabled:
+          deprecated: true
           type: boolean
+          description: "Deprecated: embedded OIDC support is now deprecated"
```
Comment (Isan-Rivkin) [inline api/swagger.yml:615]: Consider adding "... and will be removed in the upcoming versions"

File: cmd/lakefs/cmd/run.go [hunk 2]
```diff
@@ -257,6 +257,9 @@ var runCmd = &cobra.Command{
 		var oauthConfig *oauth2.Config
 		var oidcProvider *oidc.Provider
 		if cfg.Auth.OIDC.Enabled {
+			logger.
+				WithField("feature", "oidc").
+				Warn("Enabling OIDC on lakeFS server, but this functionality is deprecated")
```
Comment (Isan-Rivkin) [inline cmd/lakefs/cmd/run.go:262]: Consider adding "... and will be removed in the upcoming versions"

File: docs/assets/js/swagger.yml [hunk 3]
```diff
@@ -246,6 +246,10 @@ components:
           enum: [common_prefix, object]
         physical_address:
           type: string
+          description: |
+            The location of the object on the underlying object store.
+            Formatted as a native URI with the object store type as scheme ("s3://...", "gs://...", etc.)
+            Or, in the case of presign=true, will be an HTTP URL to be consumed via regular HTTP GET
```
Comment (Isan-Rivkin) [inline docs/assets/js/swagger.yml:252]: Is it related to deprecation?

File: docs/assets/js/swagger.yml [hunk 4]
```diff
@@ -874,6 +882,12 @@ components:
           type: boolean
         upgrade_url:
           type: string
+    GarbageCollectionConfig:
+      type: object
+      properties:
+        grace_period:
+          description: Duration in seconds. Objects created in the recent grace_period will not be collected.
+          type: integer
```
Comment (Isan-Rivkin) [inline docs/assets/js/swagger.yml:890]: Related to deprecation?

File: docs/assets/js/swagger.yml [hunk 5]
```diff
@@ -3868,6 +3879,21 @@ paths:
...
+      description: get information of gc settings
+      responses:
+        200:
+          description: lakeFS garbage collection config
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/GarbageCollectionConfig"
+        401:
+          $ref: "#/components/responses/Unauthorized"
```
Comment (Isan-Rivkin) [inline docs/assets/js/swagger.yml:3896]: Is this related in anyway to the OIDC deprecation?

**Review by Isan-Rivkin — CHANGES_REQUESTED** [review]

File: api/swagger.yml (hunk 1 above)
Comment (arielshaqed) [reply api/swagger.yml:615]: Not done: this is a product decision.  Our [communication on CHANGELOG](https://github.com/treeverse/lakeFS/blob/bf692ea20db7e9f9f48ef646a3eb36afbed53c33/CHANGELOG.md?plain=1#L7) is merely "deprecated".

File: docs/assets/js/swagger.yml (hunk 3 above)
Comment (arielshaqed) [reply docs/assets/js/swagger.yml:252]: No, just a missing "make gen".  I'll rebase before pushing.  The automated generated copy of Swagger.yml must be generated automatically.

File: docs/assets/js/swagger.yml (hunk 4 above)
Comment (arielshaqed) [reply docs/assets/js/swagger.yml:890]: As above.

**Review by arielshaqed — COMMENTED** [review]
Thanks!

Didn't do either thing (add to deprecation, fix automatic copies) as explained inside.

**Review by arielshaqed — COMMENTED** [review]
Other changes to swagger.yml copy went away after rebasing  🎆

File: api/swagger.yml (hunk 1 above)
Comment (ozkatz) [reply api/swagger.yml:615]: If we expect OIDC to remain in lakeFS cloud and for lakeFS cloud to be compatible with existing clients, I would suggest not marking it as deprecated in the API itself (docs changes that were already pushed should clarify and suffice)

**Review by ozkatz — CHANGES_REQUESTED** [review]
I believe this PR should be closed since the deprecated features are only deprecated for the lakeFS OSS server. Clients should still support this functionality.

File: api/swagger.yml (hunk 1 above)
Comment (arielshaqed) [reply api/swagger.yml:615]: 1. It is hard to imagine a client using getSetupState _and_ a generated OpenAPI client.  webui does use the endpoint, but needs to change and anyway doesn't use a generated client.
1. If there is such a client, it probably doesn't use `oidc_enabled` -- and will indeed be unaffected by this change.
1. If there is such a client that _does_ use `oidc_enabled` then it will stop working, because this property is indeed going away in lakeFS and also in Cloud.

Comment (ozkatz) [reply api/swagger.yml:615]: Thanks for the context. I agree it should be marked as deprecated then.

**Review by ozkatz — APPROVED** [review]

**Review by Isan-Rivkin — APPROVED** [review]

**Comment by arielshaqed** [comment]
Thanks!

Rebased on trunk, pulling once it's green again.
